solidity/test/isms/PolygonPosIsm.t.sol

```diff
@@ -0,0 +1,365 @@
+
+    uint256 internal mainnetFork;
+    uint256 internal polygonPosFork;
+
+    address internal constant POLYGON_CROSSCHAIN_SYSTEM_ADDR =
+        0x0000000000000000000000000000000000001001;
+
+    address internal constant MUMBAI_FX_CHILD =
+        0xCf73231F28B7331BBe3124B907840A94851f9f11;
+    address internal constant GOERLI_CHECKPOINT_MANAGER =
+        0x2890bA17EfE978480615e330ecB65333b880928e;
+    address internal constant GOERLI_FX_ROOT =
+        0x3d1d3E34f7fB6D26245E6640E1c50710eFFf15bA;
+
+    address internal constant MAINNET_FX_CHILD =
+        0x8397259c983751DAf40400790063935a11afa28a;
+    address internal constant MAINNET_CHECKPOINT_MANAGER =
+        0x86E4Dc95c7FBdBf52e33D563BbDB00823894C287;
+    address internal constant MAINNET_FX_ROOT =
+        0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2;
+    address internal constant MAINNET_STATE_SENDER =
+        0x28e4F3a7f651294B9564800b2D01f35189A5bFbE;
+
+    uint8 internal constant POLYGON_POS_VERSION = 0;
+    uint8 internal constant HYPERLANE_VERSION = 1;
+
+    TestMailbox internal l1Mailbox;
+    PolygonPosIsm internal polygonPosISM;
+    PolygonPosHook internal polygonPosHook;
+    FxChild internal fxChild;
+
+    TestRecipient internal testRecipient;
+    bytes internal testMessage =
+        abi.encodePacked("Hello from the other chain!");
+    bytes internal testMetadata =
+        StandardHookMetadata.overrideRefundAddress(address(this));
+
+    bytes internal encodedMessage;
+    bytes32 internal messageId;
+
+    uint32 internal constant MAINNET_DOMAIN = 1;
+    uint32 internal constant POLYGON_POS_DOMAIN = 137;
+
+    event StateSynced(
+        uint256 indexed id,
+        address indexed contractAddress,
+        bytes data
+    );
+
+    event ReceivedMessage(bytes32 indexed messageId);
+
+    function setUp() public {
+        // block numbers to fork from, chain data is cached to ../../forge-cache/
+        mainnetFork = vm.createFork(vm.rpcUrl("mainnet"), 18_718_401);
+        polygonPosFork = vm.createFork(vm.rpcUrl("polygon"), 50_760_479);
+
+        testRecipient = new TestRecipient();
+
+        encodedMessage = _encodeTestMessage();
+        messageId = Message.id(encodedMessage);
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            SETUP                            ///
+    ///////////////////////////////////////////////////////////////////
+
+    function deployPolygonPosHook() public {
+        vm.selectFork(mainnetFork);
+
+        l1Mailbox = new TestMailbox(MAINNET_DOMAIN);
+
+        polygonPosHook = new PolygonPosHook(
+            address(l1Mailbox),
+            POLYGON_POS_DOMAIN,
+            TypeCasts.addressToBytes32(address(polygonPosISM)),
+            MAINNET_CHECKPOINT_MANAGER,
+            MAINNET_FX_ROOT
+        );
+
+        polygonPosHook.setFxChildTunnel(address(polygonPosISM));
+
+        vm.makePersistent(address(polygonPosHook));
```

Comment:
what does this do?

Reply:
hook on Ethereum inherents FxRootTunnel and need to set a destination, which is FxChildTunnel on Polygon Pos